ReactSkia/views/common/RSkDrawUtils.cpp

```diff
@@ -364,43 +368,58 @@ bool  drawShadow(SkCanvas *canvas,Rect frame,
            shadowOn = Border;
     }
     if(shadowOn != None) {
-        SkRect clipRect = SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height);
-        SkVector radii[4]={{borderProps.borderRadii.topLeft,borderProps.borderRadii.topLeft},
-                       {borderProps.borderRadii.topRight,borderProps.borderRadii.topRight}, \
-                       {borderProps.borderRadii.bottomRight,borderProps.borderRadii.bottomRight }, \
-                       {borderProps.borderRadii.bottomLeft,borderProps.borderRadii.bottomLeft}};
-        if(borderProps.borderWidths.left) {
-            clipRect.inset(borderProps.borderWidths.left/2,borderProps.borderWidths.left/2);
+        SkPaint paint;
+        if(componentShadow.shadowRadius != 0) {
+           if(componentShadow.maskFilter == nullptr) {
+              componentShadow.createMaskFilter();
+            }
+            paint.setMaskFilter(componentShadow.maskFilter);
         }
+        bool needsSaveLayer=(componentShadow.shadowOpacity|| (shadowOn == Background));
```

Comment:
condition to be changed to
 bool needsSaveLayer=(componentShadow.shadowOpacity|| ((shadowOn == Background) && !shadowOnTransparentComponent));